src/main/kotlin/systems/ajax/malov/stockanalyzer/repository/impl/MongoStockRecordRepository.kt

```diff
@@ -27,26 +25,48 @@ import org.springframework.data.mongodb.core.mapping.Document as SpringDocument
 
 @Repository
 class MongoStockRecordRepository(
-    private val mongoTemplate: MongoTemplate,
+    private val reactiveMongoTemplate: ReactiveMongoTemplate,
 ) : StockRecordRepository {
-    override fun insertAll(mongoStockRecords: List<MongoStockRecord>): List<MongoStockRecord> {
-        return mongoTemplate.insertAll(mongoStockRecords).toList()
+    override fun insertAll(mongoStockRecords: List<MongoStockRecord>): Flux<MongoStockRecord> {
+        return reactiveMongoTemplate.insertAll(mongoStockRecords)
     }
 
-    override fun findAllStockSymbols(): List<String> {
-        return mongoTemplate.getCollection(MongoStockRecord.COLLECTION_NAME)
-            .distinct(MongoStockRecord::symbol.name, String::class.java)
-            .toList()
+    override fun findAllStockSymbols(): Mono<List<String>> {
+        return reactiveMongoTemplate.query(MongoStockRecord::class.java)
+            .distinct(MongoStockRecord::symbol.name)
+            .asType<String>()
+            .all()
+            .collectList()
     }
 
     override fun findTopNStockSymbolsWithStockRecords(
         n: Int,
         from: Date,
         to: Date,
-    ): Map<String, List<MongoStockRecord>> {
-        val maxChange = getMaxBigDecimal(MongoStockRecord::change.name, from, to)
-        val maxPercentChange = getMaxBigDecimal(MongoStockRecord::percentChange.name, from, to)
+    ): Mono<Map<String, List<MongoStockRecord>>> {
+        val maxChangeMono = getMaxBigDecimal(MongoStockRecord::change.name, from, to)
+        val maxPercentChangeMono = getMaxBigDecimal(MongoStockRecord::percentChange.name, from, to)
+
+        return Mono.zip(maxChangeMono, maxPercentChangeMono)
+            .flatMap { maxValues ->
+                val (maxChange, maxPercentChange) = maxValues
```

Comment:
can we rewrite:
```suggestion
            .flatMap { (maxChange, maxPercentChange) ->
```
?